src/host-config/types.ts

```diff
@@ -61,9 +73,18 @@ export type HostConfiguration = t.TypeOf<typeof HostConfiguration>;
  * valid but has the correct shape.
  */
 export const LegacyHostConfiguration = t.type({
-	network: t.partial({
-		proxy: t.record(t.string, t.any),
-		hostname: t.string,
-	}),
+	network: t.exact(
+		t.partial({
+			proxy: t.intersection([
+				t.record(t.string, t.any),
```

Comment:
For another PR, I think this record could be a more relaxed version of the proxy config that accepts a `string` for the type instead of a set of values